### PR TITLE
Optimize docker compilation time

### PR DESCRIPTION
**Pull request by HongjiangHuang**
Better use of docker layer cache

### CODE REVIEW

File: openapi/openapi-generator/Dockerfile [hunk 1]
```diff
@@ -56,6 +40,21 @@ RUN chown root:root /etc/apt/sources.list.d/microsoft-prod.list
 RUN apt-get update
 RUN apt-get install -yy -q dotnet-hosting-2.0.8
 
+ARG OPENAPI_GENERATOR_COMMIT
```
Comment (brendandburns) [inline openapi/openapi-generator/Dockerfile:43]: Does moving the ARGs affect the caching? I feel like it is clearer if they are at the top.

**Comment by brendandburns** [comment]
One small, otherwise looks good.

File: openapi/openapi-generator/Dockerfile (hunk 1 above)
Comment (HongjiangHuang) [reply openapi/openapi-generator/Dockerfile:43]: According to the documentation, when the ARG layer is changed, all the following layers of the cache will invalidate.


### Reference: 
https://docs.docker.com/engine/reference/builder/#impact-on-build-caching
https://stackoverflow.com/questions/41593135/does-putting-arg-at-top-of-dockerfile-prevent-layer-re-use

Comment (HongjiangHuang) [reply openapi/openapi-generator/Dockerfile:43]: @brendanburns

Comment (brendandburns) [reply openapi/openapi-generator/Dockerfile:43]: ok, thanks for the explanation.

**Pull request merged**